### ISSUE TITLE
Error when initializing struct with cfg field in multiple blocks

### ISSUE THREAD

**Issue by leudz** [rust-analyzer/rust-analyzer#3535]
Hi! I have a few structs with fields only present if some feature is set.
Rust-analyzer gives me a `missing structure fields` error when I initialize the version without all fields.
`cargo check` doesn't emit any error and it compiles just fine though.
This error only shows up if I use two blocks.

[Playground with two blocks](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3d2a29ca6f1ae825cb047c1e0017f637)
[Playground with "inlined" cfg](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7d150f8b9972b7c867fe312b32b50302)
Whether `_field1` is there or not doesn't change anything.

**Comment by jan-auer** [comment]
Also running into this. Here is a small repro:

```rust
#[derive(Debug)]
struct S {
    #[cfg(feature = "val")]
    val: u32,
}

fn main() {
    #[cfg(feature = "val")]
    let s = S { val: 42 };
    #[cfg(not(feature = "val"))]
    let s = S {};

    println!("{:?}", s);
}
```

Rust Analyzer reports this issue:

```
Missing structure fields:
- val
```

**Comment by lnicola** [comment]
This is now working correctly:

![image](https://user-images.githubusercontent.com/308347/105536446-0840ab00-5cf9-11eb-9d8b-a36e196bb3c0.png)

![image](https://user-images.githubusercontent.com/308347/105536468-1393d680-5cf9-11eb-99ab-966037e57396.png)

![image](https://user-images.githubusercontent.com/308347/105536477-18588a80-5cf9-11eb-9998-1b83f72c7535.png)
